### PR TITLE
Add new rule for seed and P2P

### PR DESCRIPTION
**Pull request by BgmLover**

### CODE REVIEW

File: src/peer/peer.cpp [hunk 1]
```diff
@@ -122,13 +118,16 @@ void Peer::ProcessVersionMessage(VersionMessage& version) {
...
+    peer_id         = versionMessage->id;
+
     char time_buffer[20];
     strftime(time_buffer, 20, "%Y-%m-%d %H:%M:%S", localtime((time_t*) &(versionMessage->nTime)));
     spdlog::info("{}: got version = {}, services = {}, time = {}, height = {}", address.ToString(),
                  versionMessage->client_version, versionMessage->local_service, std::string(time_buffer),
                  versionMessage->current_height);
 
-    if (versionMessage->current_height > DAG->GetBestMilestoneHeight()) {
+    bool compareHeight = isSeed || amISeed_;
```
Comment (bassland) [inline src/peer/peer.cpp]:
```suggestion
    bool compareHeight = !(isSeed || amISeed_);
```

File: src/config.h [hunk 2]
```diff
@@ -274,6 +282,7 @@ class Config {
     uint16_t bindPort_       = defaultPort;
     std::string connect_;
     std::string networkType_ = "Testnet";
+    bool amISeed_             = false;
```
Comment (0ersted) [inline src/config.h]: format

File: src/peer/peer.h [hunk 3]
```diff
@@ -169,28 +177,14 @@ class Peer {
      */
     void ProcessVersionACK();
 
-    /*
-     * process address message, check, relay and save addresses
-     * @param addressMessage
-     */
-    void ProcessAddressMessage(AddressMessage& addressMessage);
+
```
Comment (0ersted) [inline src/peer/peer.h]: I would remove this line here

File: src/peer/peer_manager.cpp [hunk 4]
```diff
@@ -169,9 +176,70 @@ void PeerManager::HandleMessage() {
             }
             auto msg_from = GetPeer(msg.first);
             if (msg_from && msg_from->IsVaild()) {
-                msg_from->ProcessMessage(msg.second);
+                switch (msg.second->GetType()) {
+                    case NetMessage::BLOCK: {
+                        auto* b   = dynamic_cast<Block*>(msg.second.release());
+                        b->source = Block::NETWORK;
```
Comment (0ersted) [inline src/peer/peer_manager.cpp]:
```suggestion
                        if (!conditon) {
                             continue;
                         }
```
to reduce the not-good-looking code of if-inside-if

**Pull request merged**